server/parameters/util/ArgsParser.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.core.server.parameters.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ArgsParser<RESULT> {
```

Comment:
We refactored the `CommandLine` class in the following ways:

- It has been renamed to `ArgsParser` to clearly describe its responsibility: to parse `args` array in the `main` file.
- The `Option` class has been extracted into its own class.
- The `CliHelp` interface has been deleted since it is superfluous.
- The `Help`  class has been extracted into a standalone class since it is shared by several classes: `SubcommandParser`, `OptionParser`, and `YamlParser`